src/machine/machine_avr.go

```diff
@@ -58,3 +58,93 @@ func (p GPIO) Get() bool {
 		return (val > 0)
 	}
 }
+
+const (
+	COM2A1 = 7
+	COM2A0 = 6
+	COM2B1 = 5
+	COM2B0 = 4
+	COM0A1 = 7
+	COM0A0 = 6
+	COM0B1 = 5
+	COM0B0 = 4
+	COM1A1 = 7
+	COM1A0 = 6
+	COM1B1 = 5
+	COM1B0 = 4
+
+	CS00 = 0
+	CS01 = 1
+	CS11 = 1
+	CS22 = 2
+
+	WGM10 = 0
+	WGM11 = 1
+	WGM20 = 0
+	WGM21 = 1
+)
+
+// InitPWM initializes the registers needed for PWM.
+func InitPWM() {
+	// use waveform generation
+	*avr.TCCR0A |= avr.TCCR0A_WGM0
+
+	// set timer 0 prescale factor to 64
+	*avr.TCCR0B |= 1 << CS01
+	*avr.TCCR0B |= 1 << CS00
+
+	// set timer 1 prescale factor to 64
+	*avr.TCCR1B |= 1 << CS11
+
+	// put timer 1 in 8-bit phase correct pwm mode
+	*avr.TCCR1A |= 1 << WGM10
+
+	// set timer 2 prescale factor to 64
+	*avr.TCCR2B |= 1 << CS22
+
+	// configure timer 2 for phase correct pwm (8-bit)
+	*avr.TCCR2A |= 1 << WGM20
+}
+
+// Configure configures a PWM pin for output.
+func (pwm PWM) Configure() {
+	if pwm.Pin < 8 {
+		*avr.DDRD |= 1 << pwm.Pin
+	} else {
+		*avr.DDRB |= 1 << (pwm.Pin - 8)
+	}
+}
+
+// Set sets the needed register values to turn on the duty cycle for a PWM pin.
```

Comment:
While correct, the whole point of this function is setting register values. Perhaps this description can be shortened to:

> Set turns on the duty cycle for a PWM pin.

Also, it is perhaps best to describe in what units this is - something like "0 means 0% on and 255 means 100% on" (if that is the case, I seem to remember some issue with the AVR in which PWM would not normally reach 100% when fully enabled).

At the same time, I think we should use 16-bit values here. This should be optimized to 8-bit values when calling PWM.Set(value) with a constant value known at compile time, but gives greater flexibility for platforms that do support more than 8 bits of PWM.